### PR TITLE
Refactor task management

### PR DESCRIPTION
**Pull request by amatiasq**
- Task are now divided in three phases (see documentation).
- Tasks don't have a worker now so multiple workers can work on the same task.
- Add work-area to handle a set of tiles where a task has to be made.
- Worker task setter will interact with the game itself.
- Improved tiles debug color rendering

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -43,11 +48,15 @@ export default class App extends React.Component<AppProps, AppState> {
       );
     });
 
+    // const tasks = [];
```
Comment (Hoijof) [inline src/app/App.tsx]: cofcof

File: src/game/renderer.ts [hunk 2]
```diff
@@ -44,9 +44,13 @@ export default class Renderer {
             // this.drawTargetCell();
         }
 
+        const tiles = this.game.getVisibleTiles();
+
+        tiles.forEach(tile => tile.renderTile(this.background));
+        tiles.forEach(tile => tile.renderEntities(this.context));
+
         // this.drawOccupiedCells();
```
Comment (Hoijof) [inline src/game/renderer.ts]: Venga comments xD

File: src/meta/best-of.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    });
+
+    return best;
+}
+
+type Serializer<T, U> = (value: T) => U;
+
+interface ForEacheable<T> {
+    forEach(iterator: (value: T) => void): void;
+}
```
Comment (Hoijof) [inline src/meta/best-of.ts]: SALTO DE LINEAAAA LERELELELRELREL

File: src/app/App.tsx (hunk 1 above)
Comment (amatiasq) [reply src/app/App.tsx]: Estaba a punto de commitearlo sin comentar pero me acordé que no va xD

File: src/game/renderer.ts (hunk 2 above)
Comment (amatiasq) [reply src/game/renderer.ts]: Si, estos ya los podríamos borrar

File: src/meta/best-of.ts (hunk 3 above)
Comment (amatiasq) [reply src/meta/best-of.ts]: 🙄

**Pull request merged**